File: src/config/presets.py
"""
Color presets and animation mode name mappings.
"""

from src.config.hardware import (
    MODE_BREATHING,
    MODE_FLASHING,
    MODE_DOUBLE_FLASHING,
    MODE_LIGHTNING,
    MODE_METEOR,
    MODE_COLOR_PULSE,
    MODE_COLOR_SHIFT,
    MODE_COLOR_WAVE,
    MODE_MARQUEE,
    MODE_RAINBOW_WAVE,
    MODE_VISOR,
    MODE_STACK,
    MODE_FIRE,
)

COLOR_PRESETS = {
    "red": (255, 0, 0),
    "green": (0, 255, 0),
    "blue": (0, 0, 255),
    "cyan": (0, 255, 255),
    "magenta": (255, 0, 255),
    "yellow": (255, 255, 0),
    "white": (255, 255, 255),
    "orange": (255, 120, 0),
    "purple": (160, 32, 240),
    "off": (0, 0, 0),
}

ANIMATION_MODES = {
    "rainbow_wave": MODE_RAINBOW_WAVE,
    "breathing": MODE_BREATHING,
    "meteor": MODE_METEOR,
    "flashing": MODE_FLASHING,
    "double_flashing": MODE_DOUBLE_FLASHING,
    "lightning": MODE_LIGHTNING,
    "color_pulse": MODE_COLOR_PULSE,
    "color_shift": MODE_COLOR_SHIFT,
    "color_wave": MODE_COLOR_WAVE,
    "marquee": MODE_MARQUEE,
    "visor": MODE_VISOR,
    "stack": MODE_STACK,
    "fire": MODE_FIRE,
}

File: src/config/presets.rs
//! Color presets and animation mode name mappings.

use super::hardware::*;

pub fn get_color_preset(name: &str) -> Option<(u8, u8, u8)> {
    match name.to_lowercase().as_str() {
        "red" => Some((255, 0, 0)),
        "green" => Some((0, 255, 0)),
        "blue" => Some((0, 0, 255)),
        "cyan" => Some((0, 255, 255)),
        "magenta" => Some((255, 0, 255)),
        "yellow" => Some((255, 255, 0)),
        "white" => Some((255, 255, 255)),
        "orange" => Some((255, 120, 0)),
        "purple" => Some((160, 32, 240)),
        "off" => Some((0, 0, 0)),
        _ => None,
    }
}

pub fn get_animation_mode(name: &str) -> Option<u8> {
    match name.to_lowercase().as_str() {
        "rainbow_wave" => Some(MODE_RAINBOW_WAVE),
        "breathing" => Some(MODE_BREATHING),
        "meteor" => Some(MODE_METEOR),
        "flashing" => Some(MODE_FLASHING),
        "double_flashing" => Some(MODE_DOUBLE_FLASHING),
        "lightning" => Some(MODE_LIGHTNING),
        "color_pulse" => Some(MODE_COLOR_PULSE),
        "color_shift" => Some(MODE_COLOR_SHIFT),
        "color_wave" => Some(MODE_COLOR_WAVE),
        "marquee" => Some(MODE_MARQUEE),
        "visor" => Some(MODE_VISOR),
        "stack" => Some(MODE_STACK),
        "fire" => Some(MODE_FIRE),
        _ => None,
    }
}
